kgl-math/src/commonMain/kotlin/com/kgl/math/Vectors.kt

```diff
@@ -0,0 +1,289 @@
+package com.kgl.math
+
+/**
+ * A generic two-component vector. Methods in this interface support only read-only access to the vector;
+ * read/write access is supported through the [MutableVector2] interface.
+ * @param T the type of component contained in the vector. The vector is invariant on its component type.
+ */
+interface Vector2<T> {
+
+	val x: T
+	val y: T
+
+	operator fun component1(): T = x
+	operator fun component2(): T = y
+
+	operator fun get(index: Int): T = when (index) {
+		0 -> x; 1 -> y; else -> throw IndexOutOfBoundsException()
+	}
+
+	fun copy(): Vector2<T>
+
+
+	// geometric
+
+	val length: T
+
+	fun normalize(): Vector2<T>
+
+	infix fun dot(other: Vector2<T>): T
+
+
+	// arithmetic
+
+	operator fun plus(scalar: T): Vector2<T>
+	operator fun minus(scalar: T): Vector2<T>
+	operator fun times(scalar: T): Vector2<T>
+	operator fun div(scalar: T): Vector2<T>
+
+	operator fun plus(other: Vector2<T>): Vector2<T>
+	operator fun minus(other: Vector2<T>): Vector2<T>
+	operator fun times(other: Vector2<T>): Vector2<T>
```

Comment:
Isn't this somewhat ambiguous? It could be cross-product or dot-product.

Reply:
none of these are cross or dot products. There are dedicated dot and cross functions for those. The arithmetic operators will be component-wise operations